spec/src/language_negotiation_lambda_spec.js

```diff
@@ -0,0 +1,160 @@
+            { cookie: "fr",     expected: "es" },
+            { cookie: "",       expected: "es" }
+        ].forEach( function( cookies ) {
+            it( `Resolves cookie ${cookies.cookie} to ${cookies.expected}`, function( done ) {
+                event.Records[0].cf.request.headers.Cookie = [
+                    process.env.LANGUAGE_COOKIE_NAME+"="+cookies.cookie
+                ];
+                event.Records[0].cf.request.headers['Accept-Language'] = ['es'];
+
+                languageNegotiationLambda.handler( event, { /* context */ }, (err, result) => {
+                    try {
+                        expect( err ).toBeNull();
+                        expect( result.headers['X-Accept-Language'] ).toEqual([cookies.expected]);
+                        done();
+                    }
+                    catch( error ) {
+                        done( error );
+                    }
+                });
+            });
+        });
+    });
+
+    describe( 'Valid headers', function(){
+        [
+            { header: "en",                   customHeader: "en" },
+            { header: "es",                   customHeader: "es" },
+            { header: "de",                   customHeader: "en" },
+            { header: "en;q=0.8,es",          customHeader: "es" },
+            { header: "es;q=0.8,en",          customHeader: "en" },
+            { header: "de,en;q=0.8",          customHeader: "en" },
+            { header: "de,es;q=0.8",          customHeader: "es" },
+            { header: "es;q=NaN",             customHeader: "es" },
+            { header: "es;q=NaN,en;q=0.01",   customHeader: "en" },
+            { header: "de,en;q=0.2,es;q=0.8", customHeader: "es" },
+            { header: "EN",                   customHeader: "en" },
+            { header: "ES",                   customHeader: "es" },
+            { header: "EN-US",                customHeader: "en" },
+            { header: "ES-419",               customHeader: "es" }
+        ].forEach( function( headers ) {
+            it( `Resolves header ${headers.header} to ${headers.customHeader}`, function( done ) {
+                event.Records[0].cf.request.headers['Accept-Language'] = [headers.header];
+
+                languageNegotiationLambda.handler( event, { /* context */ }, (err, result) => {
+                    try {
+                        expect( err ).toBeNull();
+                        expect( result.headers['X-Accept-Language'] ).toEqual([headers.customHeader]);
+                        done();
+                    }
+                    catch( error ) {
+                        done( error );
+                    }
+                });
+            });
+        });
+    });
+
+    describe( 'Invalid cookies', function(){
+        [
+            undefined,
+            NaN,
+            null,
+            {}
+        ].forEach( function( cookie ) {
+            it( `Passes on cookie ${cookie} and uses browser language`, function( done ) {
```

Comment:
I think using something like `ignores cookie` would be clearer here. `passes on` could be interpreted as `fowards`, which might be confusing.